### ISSUE TITLE
task tax rate not added and typing in a "product" name doesn't assign tax to task - pick list needed

### ISSUE THREAD

**Issue by oats197** [invoiceninja/flutter-mobile#60]
So I've set my tax rates in settings on the self hosted server side Settings/Advanced Settings to 
"Enable specifying line item taxes"  Nothing else checked.  Default Tax Rate is set to 13%

I'm confused about how to set a tax rate for a task. Clients have different "task rates" specified in the Client section.  From a computer (not phone app), invoicing task(s) works but I have to change the product of the task line to "Labour" from blank, to assign a tax rate. Everything here is 13% tax except a line item aka "product" for environmental recovery surcharge on hardware - this varies from monitor to desktop to laptop to mouse etc. but is not "taxed" as well.

I can continue to do the assigning a "Labour" product to the tasks - if I miss that no tax is collected - so not great - if you have a suggestion I'm open to it at this level.

This problem leads to the problem on my android phone InvNinja app...  If I create an invoice with the task(s) performed at a job site, It allows me to click the "Product" assignment and type something in, but no pick list.  If I type Labour or labour, it doesn't assign a tax even though Labour is a valid product with a tax rate assigned.

I see in Advanced Settings / Invoice Settings / Tasks two entries for "Task Field"  I'm unclear if this is something I should set up or not related to this issue of just assigning a tax to tasks which would solve the issue.
Other possible solutions - can a task just be always affiliated with a "Product"?

"specifying an invoice tax" doesn't work because I can't subtract tax for those line items for "Electronics Stewardship Fees"

Basically, I can't invoice tasks from my phone because of this issue / or lack of understanding on my backend setup.  Your guidance would be greatly appreciated.

**Comment by hillelcoren** [comment]
Are you able to manually select the tax rate in the mobile app?

**Comment by oats197** [comment]
No option for that. Just product, description, unit cost, quantity. I can
edit any of those lines. But don't see any way to you at a tax.

**Comment by hillelcoren** [comment]
Which version of the app are you using? You can check by clicking 'About' in the menu. 

Also, have you tried clicked 'Refresh Data' in the mobile app settings.

**Comment by oats197** [comment]
Phone app version is 1.41
Refresh in settings didn't change anything. Still no option to pick a tax.
Manually typing in product called labour. Doesn't assign any tax.

**Comment by hillelcoren** [comment]
I'm not sure, line item taxes are working for other users. 

Can you post a screenshot of the tax settings in the web app and the edit invoice item screen in the mobile app.

**Comment by oats197** [comment]
![AndroidAppCreateInvoiceScreen](https://user-images.githubusercontent.com/47153718/56098934-6578bd00-5ed4-11e9-9a19-08ee5e3add74.png)
![TaxSettingsScreen](https://user-images.githubusercontent.com/47153718/56098935-66115380-5ed4-11e9-903a-f2babaf218d3.png)

**Comment by oats197** [comment]
Maybe it's my setup at the web app side?  When I create a task(s) and then invoice the task(s), it doesn't assign a product to it or a tax.  I have to pick a product manually and the 13% shows up OR I can pick a tax from the drop list under "tax".  Have I done something wrong that doesn't auto add tax to the line item on the server/web app side maybe?

Also, from the android app.  I can pick a product to add to an invoice - that's fine -- tax shows there.  But adding tasks = no tax.

**Comment by hillelcoren** [comment]
Thanks for the details!

If you add the task and then edit the line item do you see the tax options?

**Comment by oats197** [comment]
Web page yes.  Android app no.  
On Android, my workflow is:
-create task or pick pre-created task when I get to a job.  Start timer.  Stop timer. Choose option to "invoice task".  Then there is no option to change a tax or a pick list for product.  I tried the cloud save button just now - no difference at that point or if I back out of the line item and going back into the same line item.  The screen is always as shown in screenshot in these cases.  Adding a line item at this point allows me to pick a product - the tax shows when I do that for "Labour" product I have.  Just not for tasks created in the tasks section.
On the web app, with the same sequence as above, then I MUST pick "Labour" from my products, it adds the tax to the task.  When I first started using InNin - I didn't remember to do that a couple of times and sent invoices without the tax added.  Luckily, my client caught the mistake.
I have different labour rates for different companies.  It's pulling the rate for task on a per company basis just fine. (love that feature by the way).  Probably not a factor but I've seen weirder / harder to trace problems so I thought I would give you that info too.  Also, in case it matters for testing, I use the Kanban to pre-create tasks.
